src/test/java/hudson/plugins/copyartifact/CopyArtifactWorkflowTest.java

```diff
@@ -146,4 +156,173 @@ private WorkflowJob createWorkflow(String name, String script) throws IOExceptio
+        );
+        jenkinsRule.assertLogContains("foobar", jenkinsRule.assertBuildStatusSuccess(copier.scheduleBuild2(0)));
+    }
+
+    @Test
+    public void testSavedBuildSelector() throws Exception {
+        WorkflowJob copiee = createWorkflow(
+            "copiee",
+            "writeFile text: 'foobar', file: 'artifact.txt';"
+            + "archive includes: 'artifact.txt';"
+        );
+        jenkinsRule.assertBuildStatusSuccess(copiee.scheduleBuild2(0)).keepLog();
+
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: latestSavedBuild()));"
+            + "echo readFile('artifact.txt');"
+        );
+        jenkinsRule.assertLogContains("foobar", jenkinsRule.assertBuildStatusSuccess(copier.scheduleBuild2(0)));
+    }
+
+    @Test
+    public void testSpecificBuildSelector() throws Exception {
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: specific(\"${build('copiee').number}\")));"
+            + "echo readFile('artifact.txt');"
+        );
+        createWorkflow(
+            "copiee",
+            "writeFile text: 'foobar', file: 'artifact.txt';"
+            + "archive includes: 'artifact.txt';"
+        );
+        jenkinsRule.assertLogContains("foobar", jenkinsRule.assertBuildStatusSuccess(copier.scheduleBuild2(0)));
+    }
+
+    @Test
+    public void testStatusBuildSelector() throws Exception {
+        WorkflowJob copiee = createWorkflow(
+            "copiee",
+            "writeFile text: 'foobar', file: 'artifact.txt';"
+            + "archive includes: 'artifact.txt';"
+        );
+        jenkinsRule.assertBuildStatusSuccess(copiee.scheduleBuild2(0));
+
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: lastSuccessful()));"
+            + "echo readFile('artifact.txt');"
+        );
+        jenkinsRule.assertLogContains("foobar", jenkinsRule.assertBuildStatusSuccess(copier.scheduleBuild2(0)));
+    }
+
+    @Test
+    public void testTriggeredBuildSelector() throws Exception {
+        WorkflowJob copiee = createWorkflow(
+            "copiee",
+            "writeFile text: 'foobar', file: 'artifact.txt';"
+            + "archive includes: 'artifact.txt';"
+            + "build('copier');"
+        );
+
+        WorkflowJob copier = createWorkflow(
+            "copier",
+            "step(copyArtifacts(projectName: 'copiee', selector: upstream()));"
+            + "echo readFile('artifact.txt');"
+        );
+        jenkinsRule.assertBuildStatusSuccess(copiee.scheduleBuild2(0));
+        jenkinsRule.assertLogContains("foobar", jenkinsRule.assertBuildStatusSuccess(copier.getLastBuild()));
+    }
+
+    @Test
+    public void testWorkspaceBuildSelector() throws Exception {
+        // workspace is useless for pipeline jobs
```

Comment:
Then delete the test.

Reply:
I actually believe `workspace` feature is useless for pipelines, but want to keep this test to ensure the symbol works.
Changed the comment to clarify the limitation of `workspace` (that is, only applicable to `AbstractProject`s).